lib/api_resources/servers.js

```diff
@@ -430,11 +430,38 @@ ServerResource.prototype.deviceAction = function(env, next) {
     var args = [body.action];
 
     if (action.fields && action.fields.length) {
-      action.fields.forEach(function(field) {
+
+      var argsOk = action.fields.every(function(field) {
```

Comment:
Makes sense. You think a simple text error will suffice? Or should we do something more complicated with a JSON response like the `query-error`?